domain/relation/state/doc.go

```diff
@@ -3,4 +3,61 @@
 
 // Package state provides the state methods used by the relation service
 // for its domain.
+//
+// Interactions with the relations database schema:
+//
+// When an application is deployed, rows are created in the `application_endpoint`
+// table to link an application, the charm's relation and a space together.
+//
+// A binding is the combination of a charm's relation and a space. Bindings
+// are used to create an instance and when the relation when is created. A
+// user updates bindings by refreshing the application or using the bind
+// command.
+//
+// During deployment of an application with a peer relation, a row is added to
+// the `relation` and `relation_endpoint` tables. A peer relation is between
+// the units of a single application and not added by a user.
```

Comment:
We should use this bit to update https://canonical-juju.readthedocs-hosted.com/en/3.6/user/reference/relation/#peer too.